scripts/test-ci-all.sh

```diff
@@ -27,6 +27,16 @@ cargo build ${CARGO_PROFILE:+--profile ${CARGO_PROFILE}} --all --all-targets
 >&2 echo "Pre-building tests..."
 cargo test --no-run ${CARGO_PROFILE:+--profile ${CARGO_PROFILE}} -p fedimint-tests
 
+function cli_test_bin_version_hash() {
+  set -eo pipefail # pipefail must be set manually again
+  trap 'echo "## FAILED: ${FUNCNAME[0]}"' ERR
+
+  echo "## START: ${FUNCNAME[0]}"
+  unshare -rn bash -c "ip link set lo up && exec unshare --user ./scripts/version-hash-tests.sh" 2>&1 | ts -s
```

Comment:
Ah, right. This runs in a nix environment which does not have `git` or `.git` available, that's why we set this env var to allow the failure. I don't think there's anything we can do about it, LOL. I mean - we could, but that would require rebuilding everything all the time, which just doesn't seem worth it.



We could skip this test when running in CI, by detecting `FEDIMINT_BUILD_...` var I guess, then people would be testing it when running it locally...

But I'm not sure if it's worth it. All in all - it's not an end of the world.

In CI we could run `nix run .#fedimint-cli version-hash` and verify that it was set etc.  but that tests something completely different (nix substituting version, no the build scripts).

Reply:
> that would require rebuilding everything all the time
I don't follow your logic here. The CI build has to check out the code, right? Can we not grab the git hash at that point and pass it through the system somehow? If the git hash is not passed in when the test is executed, fall back to calling git directly, which should take care of the dev env use case.